src/rgw/rgw_rados.cc

```diff
@@ -5848,21 +5852,30 @@ int RGWRados::select_new_bucket_location(RGWUserInfo& user_info, const string& z
 
   /* now check that tag exists within zonegroup */
   /* find placement rule. Hierarchy: request rule > user default rule > zonegroup default rule */
-  string rule = request_rule;
-  if (rule.empty()) {
-    rule = user_info.default_placement;
-    if (rule.empty())
-      rule = zonegroup.default_placement;
+  const string *rule = &request_rule;
+  if (rule->empty()) {
+    rule = &user_info.default_placement;
+    if (rule->empty())
+      rule = &zonegroup.default_placement;
   }
 
-  if (rule.empty()) {
+  if (rule->empty()) {
     ldout(cct, 0) << "misconfiguration, should not have an empty placement rule name" << dendl;
     return -EIO;
   }
 
-  map<string, RGWZoneGroupPlacementTarget>::iterator titer = zonegroup.placement_targets.find(rule);
+  auto titer = zonegroup.placement_targets.find(*rule);
+
+  // if the given placement target doesn't exist in this zonegroup, fall back
+  // to the default placement
+  if (titer == zonegroup.placement_targets.end() &&
```

Comment:
This will violate the order of select placement rule:

```
   /* find placement rule. Hierarchy: request rule > user default rule > zonegroup default rule */
```

when request rule specified but doesn't exist, and user default rule specified and exist. The expected rule should be user default rule, not the zonegroup default rule.